### PR TITLE
Add an ability to create full clone instead of snapshot

### LINKED ISSUE #78: [FEAT] "clone" as a "snapshot"
**Is your feature request related to a problem? Please describe.**

Both Cinder Volume and Manila Share snapshots make unavailable the source resource to be deleted. Causing PV/PVC k8s resources to stuck on deletion.

**Describe the solution you'd like**

I propose an option (or a different plugin key, e.g. a new snapshot driver name with the `-clone` suffix at the end) which will make a full Volume/Share clone. This will allow not to lock the source share from being deleted.

### PR DESCRIPTION
**Pull request by kayrus**
Resolves #78

### CODE REVIEW

**Comment by kayrus** [comment]
I think it makes sense to rename the `clone: "true"` to `method: clone`. The reason behind this:
* cinder supports volume backups and upload volume to glance image methods
* manila will support share backup feature at some point

**Comment by kayrus** [comment]
I think I'll split this PR into a couple: metadata, configurable timeouts and new snapshot methods

**Comment by kayrus** [comment]
@Lirt ready for review

**Comment by Lirt** [comment]
Hi @kayrus, the code looks great as usual.

I spent a while comparing the Manila snapshot and clone method. As far as I understand Manila doesn't have clone operation supported in backend and the difference there is that `clone` method will create snapshot and immediately create share from it and then remove snapshot while `snapshot` method will keep only snapshot of the share. Is it correct?

I added one comment regarding review, but it's (again) only about log message.

**Comment by kayrus** [comment]
> Is it correct?

correct.

File: src/cinder/block_store.go [hunk 1]
```diff
@@ -169,6 +195,66 @@ func (b *BlockStore) CreateVolumeFromSnapshot(snapshotID, volumeType, volumeAZ s
...
 
+func (b *BlockStore) createVolumeFromClone(cloneID, volumeType, volumeAZ string) (string, error) {
+	logWithFields := b.log.WithFields(logrus.Fields{
+		"cloneID":       cloneID,
+		"volumeType":    volumeType,
+		"volumeAZ":      volumeAZ,
+		"cloneTimeout":  b.cloneTimeout,
+		"volumeTimeout": b.volumeTimeout,
+	})
+	logWithFields.Info("BlockStore.CreateVolumeFromSnapshot called")
```
[the remark this answers is not in the document]
Comment (kayrus) [reply src/cinder/block_store.go:208]: Log field looks more cleaner.

[the remark this answers is not in the document]
Comment (Lirt) [reply src/cinder/block_store.go:208]: Thank you

**Review by Lirt — APPROVED** [review]

**Comment by Lirt** [comment]
Ready for merge?

**Comment by kayrus** [comment]
@Lirt yes

**Pull request merged**